src/compiler/mod.rs

```diff
@@ -219,8 +219,18 @@ impl Visitor<Value> for Compiler {
         val
     }
 
-    fn visit_unary(&mut self, _: &expression::Unary) -> Value {
-        todo!()
+    fn visit_unary(&mut self, expr: &expression::Unary) -> Value {
+        match expr.operator {
+            expression::Operator::Minus => {
+                let r = match self.walk(&expr.right) {
+                    Value::Numeric(p) => p,
+                    _ => panic!("panic"),
+                };
+
+                Value::Numeric(self.builder.build_fneg(r, ""))
+            }
+            _ => todo!("{:?}", expr.operator),
```

Comment:
Since we don't have any other operators to support here for now, we can just use regular panic like `panic!("operator {:?} is not valid for unary operations", expr.operator)`